metricbeat/docs/modules/containerd.asciidoc

```diff
@@ -0,0 +1,92 @@
+////
+This file is generated! See scripts/mage/docs_collector.go
+////
+
+:modulename: containerd
+
+[[metricbeat-module-containerd]]
+[role="xpack"]
+== Containerd module
+
+beta[]
+
+include::{libbeat-dir}/shared/integration-link.asciidoc[]
+
+:modulename!:
+
+Containerd module collects cpu, memory and blkio statistics about
+running containers controlled by containerd runtime.
+
+The current metricsets are: `cpu`, `blkio` and `memory`. They are not enabled by default.
+
+[float]
+=== Prerequisites
+`Containerd` daemon has to be configured to provide metrics before enabling containerd module.
+
+In the configuration file located in `/etc/containerd/config.toml` metrics endpoint needs to
+be set and containerd daemon needs to be restarted.
+
+```
+[metrics]
+    address = "127.0.0.1:1338"
+```
+
+[float]
+=== Compatibility
+
+The Conatinerd module is tested with the following versions of Containerd:
+v1.5.2
+
+[float]
+=== Module-specific configuration notes
+
+For cpu metricset if `calcpct.cpu` setting is set to true, cpu usage percentages will be calculated
+and more specifically fields `containerd.cpu.usage.total.pct`, `containerd.cpu.usage.kernel.pct`, `containerd.cpu.usage.user.pct`.
+Default value is true.
+
+For memory metricset if `calcpct.memory` setting is set to true, memory usage percentages will be calculated
```

Comment:
The thought was initiated by @ChrsMark [comment](https://github.com/elastic/beats/pull/29247#discussion_r761951922). In general I also agree because those extra calculations (as well as extra iteration of the events because of them) may overload the system in case of too many containers. So it is safer to have it configurable. By default it is true anyway.